### PR TITLE
update: updated trust store implementation

### PR DESCRIPTION
**Pull request by patrickzheng200**
This PR targets the dev-rc.1 branch.
1. TrimSpace of storeType, namedStore and cert fileName.
2.  Show all the certificates when a file contains a certificate chain.

Signed-off-by: Patrick Zheng <patrickzheng@microsoft.com>

### CODE REVIEW

File: cmd/notation/cert.go [hunk 1]
```diff
@@ -440,8 +457,8 @@ func printCerts(root string) error {
 	})
 }
 
-// showRootCA displays details of a root certificate
-func showRootCA(cert *x509.Certificate) {
+// showCert displays details of a certificate
+func showCert(cert *x509.Certificate) {
 	fmt.Println("Issuer:", cert.Issuer)
```
Comment (chloeyin) [inline cmd/notation/cert.go:462]: Do we need to use tabwriter to print the result?  For example: `ioutil.PrintPlugins`

File: cmd/notation/cert.go [hunk 2]
```diff
@@ -440,8 +457,8 @@ func printCerts(root string) error {
 	})
 }
 
-// showRootCA displays details of a root certificate
-func showRootCA(cert *x509.Certificate) {
+// showCert displays details of a certificate
+func showCert(cert *x509.Certificate) {
```
Comment (chloeyin) [inline cmd/notation/cert.go:461]: Do we need to use `tabwriter` to print the result?  For example: `ioutil.PrintPlugins`

File: cmd/notation/cert.go [hunk 3]
```diff
@@ -341,17 +345,26 @@ func showCerts(opts *certShowOpts) error {
...
+
+	//write out
+	fmt.Println("Display certificate details. Starting from leaf certificate if it's a certificate chain.")
+	fmt.Println("-----------------------------------------------------------------------------------------")
+	for ind, cert := range certs {
+		showCert(cert)
+		if ind != len(certs)-1 {
+			fmt.Println("-----------------------------------------------------------------------------------------")
+		}
+	}
```
Comment (chloeyin) [inline cmd/notation/cert.go:357]: We can add a function like `DisplayCerts` for showing certs.

Comment (chloeyin) [inline cmd/notation/cert.go:357]: Do you think we need to wrap these to a function like `DisplayCerts`?

Comment (patrickzheng200) [reply cmd/notation/cert.go:357]: Oh yeah, sure. will update.

File: cmd/notation/cert.go (hunk 2 above)
Comment (patrickzheng200) [reply cmd/notation/cert.go:461]: Interesting, didn't know we have this. Let me try!

Comment (patrickzheng200) [reply cmd/notation/cert.go:466]: I found that tabWriter writes horizontally, which does not work very well for long fields such as cert.PublicKey.  Will stick to vertical list format.

File: cmd/notation/cert.go [hunk 4]
```diff
@@ -179,9 +179,9 @@ func certGenerateTestCommand(opts *certGenerateTestOpts) *cobra.Command {
 
 func addCert(opts *certAddOpts) error {
 	storeType := opts.storeType
-	namedStore := opts.namedStore
+	namedStore := strings.TrimSpace(opts.namedStore)
```
Comment (binbin-li) [inline cmd/notation/cert.go:182]: what about `storeType`?

Comment (binbin-li) [reply cmd/notation/cert.go:182]: looks like you check it in `AddCertCore`, but the `errorSlice` would be a slice of same errors. I feel we can break it early.

File: cmd/notation/cert.go [hunk 5]
```diff
@@ -316,17 +320,17 @@ func listCerts(opts *certListOpts) error {
 }
 
 func showCerts(opts *certShowOpts) error {
-	storeType := opts.storeType
+	storeType := strings.TrimSpace(opts.storeType)
```
Comment (binbin-li) [inline cmd/notation/cert.go:323]: I might be wrong, looks like empty check on `storeType` and `namedStore` is required in all commands. Maybe we can move it to a method of `certShowOpts`.

File: cmd/notation/cert.go (hunk 4 above)
Comment (patrickzheng200) [reply cmd/notation/cert.go:182]: Nice catch! Checking storeType as well. As for AddCertCore, since it's export, will need to check storeType again.

**Review by binbin-li — APPROVED** [review]
lgtm

**Pull request merged**